### ISSUE TITLE
It takes quite long time to get to a PowerShell command prompt

### ISSUE THREAD

**Issue by liykfrank** [Azure/azure-powershell#17484]
### Description

I'm working on a disconnected  Azure Stack Hub.
We have created a vm on this disconnected ASH, and installed az module on it.
Then when we tried to open a PowerShell session it took around 4 minutes.
Further more, even just run "Enable-AzureRmAlias -Scope LocalMachine -ErrorAction SilentlyContinue" still took around 20 seconds.
I guess this was something caused by CRL(certificate-revocation-list) checking.
so I tried to disable those checking by editing machine.config,
`   <system.net>
        <settings>
           <servicePointManager checkCertificateName="false" checkCertificateRevocationList="false" />
        </settings> 
   </system.net>
   <runtime>
    <generatePublisherEvidence enabled="false" />
  </runtime>
</configuration>`
But still doesn't work.
Would you please advise how to make powreshell az module work with no internet connection?

**Comment by dingmeng-xue** [comment]
@liykfrank , if you just open PowerShell session, the time cost of loading is irrelevant to modules because module should not be imported at that moment. Have you defined profile that runs when PowerShell starts? If yes, please share the profile to us.

**Comment by liykfrank** [comment]
@dingmeng-xue , Hi Dingmeng, thank you for your quick reply. 
Please find my profile file in blow
`#Begin Azure PowerShell alias import
Import-Module Az.Accounts -ErrorAction SilentlyContinue -ErrorVariable importError

if ($importerror.Count -eq 0) { 
    Enable-AzureRmAlias -Module Az.Accounts, Az.Aks, Az.AnalysisServices, Az.ApiManagement, Az.ApplicationInsights, Az.Automation, Az.Backup, Az.Batch, Az.Billing, Az.Cdn, Az.CognitiveServices, Az.Compute, Az.Compute.ManagedService, Az.ContainerInstance, Az.ContainerRegistry, Az.DataFactory, Az.DataLakeAnalytics, Az.DataLakeStore, Az.DataMigration, Az.DeviceProvisioningServices, Az.DevSpaces, Az.Dns, Az.EventGrid, Az.EventHub, Az.FrontDoor, Az.HDInsight, Az.IotCentral, Az.IotHub, Az.KeyVault, Az.LogicApp, Az.MachineLearning, Az.ManagedServiceIdentity, Az.ManagementPartner, Az.Maps, Az.MarketplaceOrdering, Az.Media, Az.Monitor, Az.Network, Az.NotificationHubs, Az.OperationalInsights, Az.PolicyInsights, Az.PowerBIEmbedded, Az.RecoveryServices, Az.RedisCache, Az.Relay, Az.Reservations, Az.ResourceGraph, Az.Resources, Az.Scheduler, Az.Search, Az.Security, Az.ServiceBus, Az.ServiceFabric, Az.SignalR, Az.Sql, Az.Storage, Az.StorageSync, Az.StreamAnalytics, Az.Subscription, Az.TrafficManager, Az.Websites -ErrorAction SilentlyContinue; 
}

#End Azure PowerShell alias import`
noting more than enable-azurermalias

**Comment by liykfrank** [comment]
Here attach the debug info, it seems the delay as caused by "sending metric". is it possible to disable this feature when we don't have internet connetion
`PS C:\Users\test>         Enable-AzureRmAlias -debug
DEBUG: 3:31:04 AM - EnableAzureRmAlias begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 3:31:06 AM - using account id 'xx@xx.com'...
DEBUG: AzureQoSEvent: CommandName - Enable-AzureRmAlias; IsSuccess - True; Duration - 00:00:04.0026872
DEBUG: Finish sending metric.
DEBUG: 3:31:29 AM - EnableAzureRmAlias end processing.`

**Comment by liykfrank** [comment]
I managed to disable the "data collection" by updating the configuration in AzurePSDataCollectionProfile.json. As in below.
![image](https://user-images.githubusercontent.com/26363251/158534769-79cf0022-aafc-406b-9015-a1e04cf87171.png)
it seems it works well. We didn't see the delay anymore.
![image](https://user-images.githubusercontent.com/26363251/158535208-82d85f54-8fcd-49f5-a2ca-3a94e884b8a0.png)
I still have 2 questions:

1. Is it safe to disable the data collection? or why do we need the "data collection"?
2. Even we have disabled the data collection, when we open a new PowerShell command prompt, it still takes several minutes. any ideas?

Thank you in advance.

**Comment by liykfrank** [comment]
Sorry for not getting back to you sooner. These days are crazy busy. Thanks a million @dingmeng-xue.
We tried to start PowerShell session without profile and execute the scripts later. 
And we notices that the following script took about 3 minutes.
`Import-Module Az.Accounts -ErrorAction SilentlyContinue -ErrorVariable importError`.

I think we can close this issue now. As we have disabled the data collection the delay only occurs when we start a new powershell session,  And once the session is ready all other operations are normal without any significant delay.
Thank you again
